daemon/containerd/image.go

```diff
@@ -94,8 +97,7 @@ func (i *ImageService) presentImages(ctx context.Context, desc containerdimages.
 		return nil, err
 	}
 	if len(presentImages) == 0 {
-		ref, _ := reference.ParseAnyReference(refOrID)
-		return nil, images.ErrImageDoesNotExist{Ref: ref}
+		return nil, errdefs.NotFound(errors.New("no matching image found"))
```

Comment:
We may have to be careful with this one; ISTR we even had to change the formatting of the `ErrImageDoesNotExist` error back to use a uppercase "No such image" because some tools depended on that to string-match errors 🙈 https://github.com/moby/moby/blob/bd1d35d72123223b9d4200ab10ce51796bf1967c/daemon/images/image.go#L34

Reply:
It gets transformed back into the ErrImageDoesNotExist by the caller:
https://github.com/moby/moby/blob/aab17154e66c982c58336c98d774f1a8e14e35dc/daemon/containerd/image_history.go#L32-L36

Also, this is replaced by the next commit with a platform not found error.

Reply:
Ah, yes, I saw the next commit removed that part. Wondering if we need to be on the safe side and prefix the "platform not found" error with `No such` or `No such image:` (and, yes, I hate it!).

Reply:
I took the message from the graphdriver implementation: https://github.com/moby/moby/blob/bd1d35d72123223b9d4200ab10ce51796bf1967c/daemon/images/image.go#L203 which doesn't have that prefix.

I think it wouldn't hurt though.. WDYT?

Reply:
Good call; I didn't check the non-containerd implementation; looking at that one;

- original error was introduced in https://github.com/moby/moby/commit/7a9cb29fb980c0ab3928272cdc24c7089b2fcf64 (https://github.com/moby/moby/pull/40725)
- moved in https://github.com/moby/moby/commit/88c02716053d3cb6f88b17ea9d4a3f21f4d8bcba (https://github.com/moby/moby/pull/41556)
- moved again in https://github.com/moby/moby/commit/50f39e724707ef121a988e422eb52045d3754802 (https://github.com/moby/moby/pull/41955)
    - related to https://github.com/docker/for-linux/issues/1170#issuecomment-750478242

First two were part of 20.10, last one is 23.0, so I think we should be good. We should check if there's no special handling in our own code (e.g. `docker run` using it to check if a pull is needed), and I guess for those string-matching CLI output, we _could_ (worst case scenario) still prepend it in the CLI when printing.

> I think it wouldn't hurt though.. WDYT?

Well, also trying to burn that stuff with fire. If it's proven that for this case we _don't_ need it, we should try to avoid it, otherwise next time we make changes, we need to ask ourselves the same question again ("was this one of those that MUST have the `No such` prefix?")

Reply:
Hmm, in this case I think we can skip adding the prefix. The graphdrivers implementation has been using this message so consumers should already expect it in this form.